.github/workflows/update_openapi_spec.yml

```diff
@@ -0,0 +1,23 @@
+name: Update OpenAPI Spec
+
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - '**/*openapi.yaml'
+      - '**/*openapi.yml'
+  workflow_dispatch:
+
+jobs:
+  update-openapi-spec:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Dispatch update spec workflow
+        uses: peter-evans/repository-dispatch@v3
+        with:
+          token: ${{ secrets.PRO_ACCESS_TOKEN }}
+          repository: localstack/openapi
+          event-type: openapi-update
+          client-payload: '{"ref": "${{ github.ref }}", "repo": "${{ github.repository }}"}'
```

Comment:
Q: Is this really about handing over the reg and repo here? This would mean that if you trigger this manually from another branch, the specs would be taken from another branch, updating the specs in the repo with a non-merged version.
Is this what we want? What would this be used for?

Reply:
Fair observation. I added the ref to the branch and the `workflow_dispatch` to be able to test the workflow from some spec changes sitting in a branch. 
In my view, this workflow should be dispatched only when a spec change is pushed on master. Mostly for testing purposes, it could be handy to have the chance to trigger it from another branch, but I can't come up with concrete examples right now. The spec won't be automatically updated anyway: a PR would still have to be approved and merged.

Reply:
Ah, right that makes sense! The additional barrier with the PR is great and should definitely be enough!
Would you mind adding a comment to the `client-payload`? I feel like this isn't obvious and might make sense to add a comment here?